gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -346,4 +346,15 @@ def self.inherited(child_class)
 
     raise "Inheriting from children of T::Enum is prohibited" if self != T::Enum
   end
+
+  # Marshal support
+  sig {params(_level: Integer).returns(String)}
+  def _dump(_level)
+    Marshal.dump(serialize)
+  end
+
+  sig {params(args: String).returns(T.attached_class)}
+  def self._load(args)
+    deserialize(Marshal.load(args))
```

Comment:
why is there a Marshal.load here?

Reply:
If I can assume that all enum values are `String`, the definition would have been the same as used in `gen-packages.rb`:

```ruby
def _dump(_levels)
  serialize
end

def self.load(args)
  deserialize(args)
end
```

However, given that the enum values can practically be anything, I decided to follow the advice in the Marshal docs: "Since ::dump outputs a string you can have _dump return a Marshal string which is Marshal.loaded in _load for complex objects."